From b1ab472f0f56146387d3822318394cb2525ad34c Mon Sep 17 00:00:00 2001
From: Thomas Stuart <tjstuart@google.com>
Date: Thu, 6 Jun 2024 22:34:33 +0000
Subject: [PATCH] enforce the calling package is the default dialer for
 VisualVoicemail

In the docs for setVisualVoicemailSmsFilterSettings, they state that
the callingPackage needs to be the default dialer. However, server side,
there is no enforcement for this. Now, every client is verified to be
the default dialer, system dialer, or carrier visual voicemail app
before changing the settings for visual voicemail.

Bug: 308932906
Test: CTS
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:c27f06d3401d8a0caf93ee539d67e687ff429f3b)
Merged-In: I951d7783f5c425c03768efb7aee7a38e299e6536
Change-Id: I951d7783f5c425c03768efb7aee7a38e299e6536
---
 src/com/android/phone/PhoneInterfaceManager.java | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index dc4290d4a..2accf86ae 100644
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -4276,10 +4276,9 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
     public void enableVisualVoicemailSmsFilter(String callingPackage, int subId,
             VisualVoicemailSmsFilterSettings settings) {
         mAppOps.checkPackage(Binder.getCallingUid(), callingPackage);
-
         enforceTelephonyFeatureWithException(callingPackage,
                 PackageManager.FEATURE_TELEPHONY_CALLING, "enableVisualVoicemailSmsFilter");
-
+        enforceVisualVoicemailPackage(callingPackage, subId);
         final long identity = Binder.clearCallingIdentity();
         try {
             VisualVoicemailSmsFilterConfig.enableVisualVoicemailSmsFilter(
@@ -4292,10 +4291,9 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
     @Override
     public void disableVisualVoicemailSmsFilter(String callingPackage, int subId) {
         mAppOps.checkPackage(Binder.getCallingUid(), callingPackage);
-
         enforceTelephonyFeatureWithException(callingPackage,
                 PackageManager.FEATURE_TELEPHONY_CALLING, "disableVisualVoicemailSmsFilter");
-
+        enforceVisualVoicemailPackage(callingPackage, subId);
         final long identity = Binder.clearCallingIdentity();
         try {
             VisualVoicemailSmsFilterConfig.disableVisualVoicemailSmsFilter(
-- 
2.46.1.824.gd892dcdcdd-goog

